docs/Architecture.md

```diff
@@ -1,42 +1,55 @@
 
-## Code structure
-Maestro codebase is organized following some concepts of Hexagonal architecture (Ports & Adapters). This "style" was used as a base of the development and had some adaptations. For example, the API is not considered a primary/input Port. Instead, it is outside the core package.
+![Management API IMAGE](./diagrams/Architecture-Management-API.jpg)
 
-Most of the business logic is present at services, and they rely on the ports to not depend on the specific implementations.
+#### Rooms API
 
-Entities are the structs that hold data on the system, and they don't provide any logic besides its initialization and validation. Therefore, services and ports should "communicate" using entities so that the entire system "speaks" the same language regarding data structs.
+Rooms API is the module that receive requests from game rooms and executes according to which kind of request was done.
 
-The API is responsible for converting the external protocol (currently gRPC with grpc-gateway) to entities, invoking the necessary service, and converting the response back.
+> ⚠ Note: The requests that Maestro receives in the Rooms API are documented in [this proto file](https://github.com/topfreegames/protos/blob/master/maestro/grpc/protobuf/events.proto).
 
-### Workers
-Maestro now provides a form of extending and adding more workers if they follow a worker interface. This was done to have the same pattern applied to the operation execution worker and runtime watcher. Workers are managed by the "WorkersManager," which starts an instance for each scheduler present on the storage. With this model, the implementation doesn't have to deal with multiple tenants (schedulers).
+![Rooms API IMAGE](./diagrams/Architecture-Rooms-API.jpg)
 
-Workers are not inside the service package, although they hold business logic about execution and flows. They were isolated since workers are a kind of service that has to implement an interface. Besides this difference, workers should be treated like core services.
+#### Operation Execution Worker
+
+> ⚠ Note: In Maestro a worker is a collection of routines that executes a flow related to one and only one **Scheduler** each.
+
+Operation Execution Worker creates for each **Scheduler** a thread that executes operations enqueued in the related **Scheduler** operation queue. So in this way became possible to track the events that happened and change a certain **Scheduler** in a healthier way.
+
+The Operations could be:
```

Comment:
I think this is not needed here, since this information is described under Operations.md#available-operations